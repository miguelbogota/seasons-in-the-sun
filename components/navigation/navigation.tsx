import Link from 'next/link';

const routes = [
  {
    path: '/',
    name: 'Game',
  },
  {
    path: '/blog',
    name: 'Blog',
  },
  {
    path: '/wiki',
    name: 'Wiki',
  },
];

export function Navigation() {
  return (
    <nav>
      {routes.map(({ path, name }) => (
        <Link key={path} href={path}>
          {name}
        </Link>
      ))}
    </nav>
  );
}
