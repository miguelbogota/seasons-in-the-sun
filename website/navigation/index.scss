@use './navigation' as navigation;

@mixin apply {
  @include navigation.apply;
}
