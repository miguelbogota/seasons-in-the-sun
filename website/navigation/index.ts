export { Navigation, type NavigationProps } from './navigation.component';
