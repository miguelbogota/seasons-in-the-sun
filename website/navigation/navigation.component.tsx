import Link from 'next/link';

export type NavigationProps = {
  // Empty for now.
};

export function Navigation(props: NavigationProps) {
  const {} = props;

  return (
    <nav className="navigation">
      {routes.map(({ path, name }) => (
        <Link key={path} href={path}>
          {name}
        </Link>
      ))}
    </nav>
  );
}

const routes = [
  {
    path: '/',
    name: 'Game',
  },
  {
    path: '/blog',
    name: 'Blog',
  },
  {
    path: '/wiki',
    name: 'Wiki',
  },
];
