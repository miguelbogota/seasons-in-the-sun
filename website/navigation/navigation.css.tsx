import { style } from '@vanilla-extract/css';

export const navigation = style({
  position: 'fixed',
  zIndex: 100,
  top: 0,
  right: 0,
  padding: '1rem',
  display: 'flex',
  justifyContent: 'flex-end',
  alignItems: 'center',
  gap: '1rem',
});
