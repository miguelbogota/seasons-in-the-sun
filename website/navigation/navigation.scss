@mixin apply {
  .navigation {
    position: fixed;
    z-index: 100;
    top: 0;
    right: 0;
    padding: 1rem;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    gap: 1rem;
  }
}
