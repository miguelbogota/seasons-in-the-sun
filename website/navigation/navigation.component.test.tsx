import { render } from '@testing-library/react';

import { Navigation } from './navigation.component';

describe('Website / Navigation / Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });
});
