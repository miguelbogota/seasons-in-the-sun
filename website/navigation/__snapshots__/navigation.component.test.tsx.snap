// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Website / Navigation / Component > should match snapshot 1`] = `
<div>
  <nav
    class="navigation_navigation__hx5tsf0"
  >
    <a
      href="/"
    >
      Game
    </a>
    <a
      href="/blog"
    >
      Blog
    </a>
    <a
      href="/wiki"
    >
      Wiki
    </a>
  </nav>
</div>
`;
